client/fs_endpoint.go

```diff
@@ -752,12 +756,13 @@ OUTER:
 				return nil
 			case <-ctx.Done():
 				return nil
-			case err, ok := <-eofCancelCh:
+			case _, ok := <-eofCancelCh:
 				if !ok {
 					return nil
 				}
 
-				return err
```

Comment:
I took a pass through `blockUntilNextLog` and aren't we swallowing non-EOF errors if we do this?

Reply:
We do. I will fix it

Reply:
I pushed a change which return err if it's not nil